src/Microsoft.Diagnostics.Monitoring.StartupHook/ParameterCapturing/Pipeline/ParameterCapturingPipeline.cs

```diff
@@ -206,6 +206,17 @@ public void SubmitRequest(StartCapturingParametersPayload payload, IFunctionProb
             }
         }
 
+        public void RequestStopAll()
+        {
+            foreach (var key in _allRequests.Keys)
```

Comment:
I think this throws an exception because the dictionary is modified during enumeration. Try iterating `_allRequests.Values`, call `StopRequest?.TrySetResult()` on each, and then clear the entire dictionary after.

You may also want to put the dictionary into a local and immediately recreate the dictionary to avoid concurrent access and then issue the stops.

Reply:
I was under the impression that ConcurrentDictionary has no problems with this because it's already snapshotting but let me double check

Reply:
I verified that this works correctly and ConcurrentDictionary effectively snapshots when iteration starts.